artcommon/artcommonlib/util.py

```diff
@@ -153,6 +153,25 @@ def get_assembly_release_date(assembly, group):
         return None
 
 
+def is_release_this_week(release_date):
+    """
+    Check if release date is in the near week
+    """
+    return datetime.strptime(release_date, "%Y-%m-%d").date() <= date.today() + timedelta(days=7)
+
+
+def get_next_release_schedule(group):
+    """
+    Get next release date based on current date
+    """
+    dev_schedules = requests.get(f'{RELEASE_SCHEDULES}/{group}.z/schedule-tasks/?flags_and__in=dev&fields=path,date_finish', headers={'Accept': 'application/json'})
```

Comment:
Maybe a retry would be good?

Reply:
from my experience, I think there is no need for a retry here